src/wasm.h

```diff
@@ -1330,6 +1330,7 @@ class Function {
 public:
   Name name;
   WasmType result;
+  int64_t indirectIndex; // pre-assigned indirect index
```

Comment:
I don't think we should add a property to the Function object in wasm.h. That represents a function in wasm in general, while this index is for the specific case of linking code from the wasm backend.

Instead, the indirect index should be kept in the wasm-linker or s2wasm code.